ipi_tests/regression_tests/runstools.py

```diff
@@ -0,0 +1,114 @@
+import subprocess as sp
+import sys
+from pathlib import Path
+import pytest
+import numpy as np
+import time
+import glob
+import os
+import shutil
+from tempfile import TemporaryDirectory
+import tempfile
+
+
+def get_info_test(parent):
+    folders = [x[0] for x in os.walk(parent)]
+    reg_tests = list()
+
+    for ff in folders:
+        if os.path.isfile(Path(ff) / "input.xml"):
+            driver_info = list()
+            try:
+                with open(Path(ff) / "driver.txt") as f:
+                    ncount = 0
+                    while True:
+                        line = f.readline()
+                        if not line:
+                            break
+                        else:
+                            driver, address, port, mode = line.split()
+                            driver_info.append((driver, address, port, mode))
+                            ncount += 1
+                    reg_tests.append((ff, driver_info))
+                    if ncount == 0:
+                        raise ValueError("driver.txt is empty")
+            except FileNotFoundError:
+                raise FileNotFoundError(
+                    "({}) An input.xml file was found but a driver.txt not".format(ff)
+                )
+            except ValueError:
+                raise ValueError(
+                    'Please specify "model address      port mode" inside driver.txt'
+                )
+
+    return reg_tests
+
+
+class Runner(object):
+    def __init__(self, parent, check_errors=True, check_prop=True):
+        self.parent = parent
+        self.check_error = check_errors
+        self.check_prop = check_prop
+
+    def _run(self, cmd1, cmd2, cwd):
+
+        try:
+            for filename in glob.glob("/tmp/ipi_*"):
```

Comment:
Note that if another test is running it may fail to produce a RESTART file in case the /tmp/Ipi_$WHATEVER file is absent.

Reply:
The tests run sequentially so this will not generate problems.
Of course, if someone is running i-pi and wants to do a regression test on the same computer, this may be inconvenient.
But,
1. This won't happen in the GitHub workflow
2. I find it much more probable that someone will try to run regression tests several times because something is failing and this function helps that task.
3. Last but not least, in the future, we will have hopefully many regression tests, so if some of them fail this function
cleans the stage for the remainder tests